### PR TITLE
Enable lint checks on cypress and build #658

### LINKED ISSUE #658: Enable lint checks for cypress and build
https://github.com/ral-facilities/datagateway/blob/54405e3687e4c37ace525efc139c9ad066285c12/packages/datagateway-dataview/package.json#L69

fix the errors

### PR DESCRIPTION
**Pull request by joshuadkitenge**
## Description

add cypress directory and yarn to the lint checker 
fix all the cypress linting issues 

## Testing instructions

Add a set up instructions describing how the reviewer should test the code

- [ ] Review code
- [ ] Check Actions build
- [ ] Review changes to test coverage
- [ ] {more steps here}

## Agile board tracking
closes #658

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -42,7 +42,7 @@
     "preview:build": "yarn build && yarn preview",
     "preview:build:dev": "yarn build --watch & yarn preview",
     "test": "vitest --coverage",
-    "lint": "eslint --max-warnings=0 --fix ./src",
+    "lint": "eslint --max-warnings=0 --fix ./src  ./cypress && yarn build",
```
Comment (joelvdavies) [inline package.json:45]:
```suggestion
    "lint": "eslint --max-warnings=0 --fix ./src  ./cypress && yarn tsc --noEmit",
```

File: cypress/tsconfig.json [hunk 2]
```diff
@@ -2,6 +2,8 @@
     "compilerOptions": {
         "strict": true,
         "target": "es5",
+        "skipLibCheck": true,
+        "noEmit": true,
```
Comment (joelvdavies) [inline cypress/tsconfig.json:6]: I would rather this be in the command rather than here.

File: package.json [hunk 3]
```diff
@@ -42,7 +42,7 @@
     "preview:build": "yarn build && yarn preview",
     "preview:build:dev": "yarn build --watch & yarn preview",
     "test": "vitest --coverage",
-    "lint": "eslint --max-warnings=0 --fix ./src",
+    "lint": "eslint --max-warnings=0 --fix ./src  ./cypress && tsc --noEmit && tsc -p cypress/tsconfig.json",
```
Comment (joelvdavies) [inline package.json:45]:
```suggestion
    "lint": "eslint --max-warnings=0 --fix ./src ./cypress && tsc --noEmit && tsc --noEmit -p cypress/tsconfig.json",
```

**Review by joelvdavies — COMMENTED** [review]
Just one comment but LGTM

File: cypress/tsconfig.json (hunk 2 above)
Comment (joshuadkitenge) [inline cypress/tsconfig.json:6]:
```suggestion
```

**Review by joelvdavies — APPROVED** [review]

**Pull request merged**